src/language.rs

```diff
@@ -1,11 +1,11 @@
-use std::collections::HashMap;
+use std::{collections::HashMap, ops::Index};
 
 use clap::ValueEnum;
 use once_cell::sync::Lazy;
 use tree_sitter::Language;
 
-#[derive(Copy, Clone, Debug, Eq, PartialEq, ValueEnum, Hash)]
-pub enum SupportedLanguageName {
+#[derive(Copy, Clone, Debug, Eq, PartialEq, ValueEnum)]
+pub enum SupportedLanguage {
```

Comment:
Ok so probably the idea would be that most of the stuff in this file would get generated by some kind of macro "spec"

`SupportedLanguage` is kind of a "token", just a simple enum that gets used as an index into all of the "all of the supported languages" collections